### PR TITLE
Release 3.0.4

### PR DESCRIPTION
**Pull request by iruzevic**
### Added
- New support page in the settings for easier access to server configuration issues.
- Options to send empty fields to entires.

### Changed
- Block Editor select option will auto close on selection.

### Fixed
- File upload validation will no longer break the API returns invalid type.

## Summary by CodeRabbit

- **New Features**
	- Added a "Support" page in settings for accessing server configuration details.
	- Included options to send empty fields to form entries.
	- Enhanced Block Editor with improved select option behavior.
	- Implemented new CSS class `.success-text` for styling success messages.
	- Updated file size labels for consistency and added helpful conversion text.
	- Improved form handling logic for conditional tags and multiple steps.
	- Prevented default form submission for better customization.

- **Bug Fixes**
	- Fixed file upload validation issue disrupting API responses.

### CODE REVIEW

File: src/Blocks/components/form/assets/form.js [hunk 1]
```diff
@@ -1322,16 +1337,22 @@ export class Form {
...
+					const response = JSON.parse(file.xhr.response);
 
-				const validationOutputKey = this.state.getStateResponseOutputKey('validation');
+					const validationOutputKey = this.state.getStateResponseOutputKey('validation');
 
-				// Output errors if ther is any.
-				if (typeof response?.data?.[validationOutputKey] !== 'undefined' && Object.keys(response?.data?.[validationOutputKey])?.length > 0) {
-					file.previewTemplate.querySelector('.dz-error-message span').innerHTML = response?.data?.[validationOutputKey]?.[file?.upload?.uuid];
-				}
+					// Output errors if ther is any.
```
Comment (piqusy) [inline src/Blocks/components/form/assets/form.js:1345]:
```suggestion
					// Output errors if there are any.
```

File: src/Troubleshooting/SettingsDebug.php [hunk 2]
```diff
@@ -283,6 +283,59 @@ public function getSettingsGlobalData(): array
...
 						],
 					],
+					[
+						'component' => 'tab',
+						'tabLabel' => \__('Support', 'eightshift-forms'),
+						'tabContent' => [
+							[
+								'component' => 'intro',
+								'introTitle' => \__('Server config center', 'eightshift-forms'),
+								'introSubtitle' => '<p>' . \__("Check these settings to check if your forms will work correctly. For more details check <a href='/wp-admin/site-health.php' target='_blank' rel='noopener noreferrer'>Site Health</a>.", 'eightshift-forms') . '</p>',
```
Comment (piqusy) [inline src/Troubleshooting/SettingsDebug.php:293]:
```suggestion
								'introSubtitle' => '<p>' . \__("Check these settings to make sure your forms will work correctly. For more details check <a href='/wp-admin/site-health.php' target='_blank' rel='noopener noreferrer'>Site Health</a>.", 'eightshift-forms') . '</p>',
```

**Review by piqusy — CHANGES_REQUESTED** [review]
Nice work 👏🏼 

Left a few wording suggestions

**Review by dadadavorin — DISMISSED** [review]
Nice 👌

**Review by kancijan — DISMISSED** [review]

File: src/Blocks/components/form/assets/form.js [hunk 3]
```diff
@@ -729,6 +729,21 @@ export class Form {
 					break;
 			}
 		}
+
+		// Used in case we have req fields with conditional tags and multiple steps.
+		if (skipFields) {
+			const skipFieldsOutput = {};
+			skipFields.forEach((skipField) => {
+				skipFieldsOutput[skipField] = this.state.getStateElementValue(skipField, formId);
+			});
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/form/assets/form.js:738]:
```suggestion
			const skipFieldsOutput = skipFields.reduce((previous, skipField) => ({
				...previous,
				[skipField]: this.state.getStateElementValue(skipField, formId),
			}), {});
```

File: src/Troubleshooting/SettingsDebug.php [hunk 4]
```diff
@@ -283,6 +283,59 @@ public function getSettingsGlobalData(): array
 							],
 						],
 					],
+					[
+						'component' => 'tab',
+						'tabLabel' => \__('Support', 'eightshift-forms'),
+						'tabContent' => [
+							[
+								'component' => 'intro',
+								'introTitle' => \__('Server config center', 'eightshift-forms'),
```
Comment (goranalkovic-infinum) [inline src/Troubleshooting/SettingsDebug.php:292]:
```suggestion
								'introTitle' => \__('Diagnostics', 'eightshift-forms'),
```

File: src/Troubleshooting/SettingsDebug.php [hunk 5]
```diff
@@ -283,6 +283,59 @@ public function getSettingsGlobalData(): array
...
+								'introSubtitle' => '<p>' . \__("Check these settings to make sure your forms will work correctly. For more details check <a href='/wp-admin/site-health.php' target='_blank' rel='noopener noreferrer'>Site Health</a>.", 'eightshift-forms') . '</p>',
```
Comment (goranalkovic-infinum) [inline src/Troubleshooting/SettingsDebug.php:293]:
```suggestion
								'introSubtitle' => '<p>' . \__("Check these settings to make sure your forms work correctly. For more details check <a href='/wp-admin/site-health.php' target='_blank' rel='noopener noreferrer'>Site Health</a>.", 'eightshift-forms') . '</p>',
```

File: src/Troubleshooting/SettingsDebug.php [hunk 6]
```diff
@@ -283,6 +283,59 @@ public function getSettingsGlobalData(): array
...
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://www.php.net/manual/en/info.configuration.php#ini.max-execution-time" target="_blank" rel="noopener noreferrer">Max execution time</a>',
+								'cardInlineSubTitle' => 'The maximum time in seconds a script is allowed to run before it is terminated by the parser.',
+								'cardInlineRightContent' => (int)\ini_get('max_execution_time') <= 30 ? '<span class="error-text">' . \ini_get('max_execution_time') . '</span>' : '<span class="success-text">' . \ini_get('max_execution_time') . '</span>',
+							],
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://www.php.net/manual/en/info.configuration.php#ini.max-input-vars" target="_blank" rel="noopener noreferrer">Max input vars</a>',
+								'cardInlineSubTitle' => 'This sets the maximum number of variables your server can use for a single function to avoid overloading your server.',
```
Comment (goranalkovic-infinum) [inline src/Troubleshooting/SettingsDebug.php:322]:
```suggestion
								'cardInlineSubTitle' => 'Maximum number of input variables that can be used in a single function.',
```

File: src/Troubleshooting/SettingsDebug.php [hunk 7]
```diff
@@ -283,6 +283,59 @@ public function getSettingsGlobalData(): array
...
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://www.php.net/manual/en/ini.core.php#ini.memory-limit" target="_blank" rel="noopener noreferrer">Memory limit</a>',
+								'cardInlineSubTitle' => 'This sets the maximum amount of memory in bytes that a script is allowed to allocate.',
+								'cardInlineRightContent' => (int)\ini_get('memory_limit') <= 64 ? '<span class="error-text">' . \ini_get('memory_limit') . '</span>' : '<span class="success-text">' . \ini_get('memory_limit') . '</span>',
+							],
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://www.php.net/manual/en/info.configuration.php#ini.max-execution-time" target="_blank" rel="noopener noreferrer">Max execution time</a>',
+								'cardInlineSubTitle' => 'The maximum time in seconds a script is allowed to run before it is terminated by the parser.',
```
Comment (goranalkovic-infinum) [inline src/Troubleshooting/SettingsDebug.php:316]:
```suggestion
								'cardInlineSubTitle' => 'Maximum time (in seconds) a script is allowed to run.',
```

File: src/Troubleshooting/SettingsDebug.php [hunk 8]
```diff
@@ -283,6 +283,59 @@ public function getSettingsGlobalData(): array
...
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://wordpress.org/documentation/article/wordpress-versions/" target="_blank" rel="noopener noreferrer">WordPress version</a>',
+								'cardInlineSubTitle' => 'The version of WordPress that is running on your server.',
+								'cardInlineRightContent' => \get_bloginfo('version'),
+							],
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://www.php.net/manual/en/ini.core.php#ini.memory-limit" target="_blank" rel="noopener noreferrer">Memory limit</a>',
+								'cardInlineSubTitle' => 'This sets the maximum amount of memory in bytes that a script is allowed to allocate.',
```
Comment (goranalkovic-infinum) [inline src/Troubleshooting/SettingsDebug.php:310]:
```suggestion
								'cardInlineSubTitle' => 'Maximum amount of memory (in bytes) that a script is allowed to allocate.',
```

File: src/Troubleshooting/SettingsDebug.php [hunk 9]
```diff
@@ -283,6 +283,59 @@ public function getSettingsGlobalData(): array
...
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://www.php.net/manual/en/info.configuration.php#ini.max-input-vars" target="_blank" rel="noopener noreferrer">Max input vars</a>',
+								'cardInlineSubTitle' => 'This sets the maximum number of variables your server can use for a single function to avoid overloading your server.',
+								'cardInlineRightContent' => (int)\ini_get('max_input_vars') <= 50 ? '<span class="error-text">' . \ini_get('max_input_vars') . '</span>' : '<span class="success-text">' . \ini_get('max_input_vars') . '</span>',
+							],
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://www.php.net/manual/en/features.file-upload.common-pitfalls.php" target="_blank" rel="noopener noreferrer">Max POST size</a>',
+								'cardInlineSubTitle' => 'The maximum size of POST data that PHP will accept.',
```
Comment (goranalkovic-infinum) [inline src/Troubleshooting/SettingsDebug.php:328]:
```suggestion
								'cardInlineSubTitle' => 'Maximum size of POST data that is accepted.',
```

File: src/Troubleshooting/SettingsDebug.php [hunk 10]
```diff
@@ -283,6 +283,59 @@ public function getSettingsGlobalData(): array
...
+								'cardInlineRightContent' => (int)\ini_get('post_max_size') <= 64 ? '<span class="error-text">' . \ini_get('post_max_size') . '</span>' : '<span class="success-text">' . \ini_get('post_max_size') . '</span>',
+							],
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://www.php.net/manual/en/features.file-upload.common-pitfalls.php" target="_blank" rel="noopener noreferrer">Max upload filesize</a>',
+								'cardInlineSubTitle' => 'The maximum size of an uploaded file.',
```
Comment (goranalkovic-infinum) [inline src/Troubleshooting/SettingsDebug.php:334]:
```suggestion
								'cardInlineSubTitle' => 'Maximum allowed file size for file uploads.',
```

File: src/Troubleshooting/SettingsDebug.php [hunk 11]
```diff
@@ -283,6 +283,59 @@ public function getSettingsGlobalData(): array
...
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://www.php.net/supported-versions.php" target="_blank" rel="noopener noreferrer">PHP version</a>',
+								'cardInlineSubTitle' => 'The version of PHP that is running on your server.',
+								'cardInlineRightContent' => \phpversion(),
+							],
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://wordpress.org/documentation/article/wordpress-versions/" target="_blank" rel="noopener noreferrer">WordPress version</a>',
+								'cardInlineSubTitle' => 'The version of WordPress that is running on your server.',
```
Comment (goranalkovic-infinum) [inline src/Troubleshooting/SettingsDebug.php:304]:
```suggestion
								'cardInlineSubTitle' => 'Version of WordPress running on the server.',
```

File: src/Troubleshooting/SettingsDebug.php [hunk 12]
```diff
@@ -283,6 +283,59 @@ public function getSettingsGlobalData(): array
...
+						'tabContent' => [
+							[
+								'component' => 'intro',
+								'introTitle' => \__('Server config center', 'eightshift-forms'),
+								'introSubtitle' => '<p>' . \__("Check these settings to make sure your forms will work correctly. For more details check <a href='/wp-admin/site-health.php' target='_blank' rel='noopener noreferrer'>Site Health</a>.", 'eightshift-forms') . '</p>',
+							],
+							[
+								'component' => 'card-inline',
+								'cardInlineTitle' => '<a href="https://www.php.net/supported-versions.php" target="_blank" rel="noopener noreferrer">PHP version</a>',
+								'cardInlineSubTitle' => 'The version of PHP that is running on your server.',
```
Comment (goranalkovic-infinum) [inline src/Troubleshooting/SettingsDebug.php:298]:
```suggestion
								'cardInlineSubTitle' => 'Version of PHP running on the server.',
```

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]

File: src/Blocks/components/form/assets/form.js (hunk 3 above)
Comment (iruzevic) [reply src/Blocks/components/form/assets/form.js:738]: I really don't like the semantics of reduce. It is hard to read and understand. Tnx for your suggestion but I will skip this, tnx

**Review by goranalkovic-infinum — APPROVED** [review]

**Pull request merged**